clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -16,12 +17,48 @@
  * This class implements a {@link LakeFSFileSystem} that can be registered to Spark and support limited write and read actions.
  */
 public class LakeFSFileSystem extends org.apache.hadoop.fs.FileSystem {
+    public static final String FS_LAKEFS_ENDPOINT = "fs.lakefs.endpoint";
+    public static final String FS_LAKEFS_ACCESS_KEY = "fs.lakefs.access.key";
+    public static final String FS_LAKEFS_SECRET_KEY = "fs.lakefs.secret.key";
+
+    private URI uri;
+    private Path workingDirectory = new Path("/");
+    private io.lakefs.clients.api.ApiClient apiClient;
 
-    @Override
     public URI getUri() {
-        return URI.create("lakefs://main");
+        return uri;
     }
 
+    @Override
+    public void initialize(URI name, Configuration conf) throws IOException {
+        super.initialize(name, conf);
+        LOG.debug("$$$$$$$$$$$$$$$$$$$$$$$$$$$$ initialize: " + name + " $$$$$$$$$$$$$$$$$$$$$$$$$$$$");
+
+        String host = name.getHost();
+        if (host == null) {
+            throw new IOException("Invalid repository specified");
+        }
+        setConf(conf);
+        this.uri = name;
+
+        // setup lakeFS api client
```

Comment:
I added these config details to https://github.com/treeverse/lakeFS/issues/1887, please add details if you want.